### PR TITLE
Added options.empty to force serialize empty inputs

### PR DESCRIPTION
**Pull request by veacks**
Hi,
Regarding the last pull request I have done a more complete one for adding empty inputs.
I have also updated the test file for that.

Hope it will be ok for you.
Thanks.
Val.

### CODE REVIEW

**Comment by veacks** [comment]
Travis doesnt seems run the unit tests, but when I `npm test` on my machine every thing seems to work fine.

File: index.js [hunk 1]
```diff
@@ -68,19 +88,34 @@ function serialize(form, options) {
...
         }
-
         result = serializer(result, key, val);
     }
 
+    // Check for all empty radio buttons and serialize them with key=""
+    if (options.empty) {
+        for (var key in radio_store) {
+            if (!radio_store[key]) {
+                result = serializer(result, key, "");
```
Comment (defunctzombie) [inline index.js]: single quotes is the norm in this file

File: index.js [hunk 2]
```diff
@@ -36,6 +37,9 @@ function serialize(form, options) {
 
     var elements = form.elements || [];
 
+    //Object store each radio and set if it's empty or not
+    var radio_store = {}
```
Comment (defunctzombie) [inline index.js]: ```var radio_store = Object.create(null);```

File: index.js [hunk 3]
```diff
@@ -57,9 +61,25 @@ function serialize(form, options) {
...
+            else if (element.checked) {
+                radio_store[element.name] = true
+            }
+        }
 
-        // value-less fields are ignored
-        if (!val) {
+        // value-less fields are ignored unless options.empty is true
+        // if options empty is true, continue only if its radio
+        if (!val && (!options.empty || (options.empty && element.type == "radio"))) {
```
Comment (defunctzombie) [inline index.js]: options.empty is checked several times in these three if statements (this one and two above). Can we do just a single if (options.empty)  ... else and do the logic for the two cases in those. Seems like that would be easier to follow.

File: index.js [hunk 4]
```diff
@@ -57,9 +61,25 @@ function serialize(form, options) {
         if ((element.type === 'checkbox' || element.type === 'radio') && !element.checked) {
             val = undefined;
         }
+        
+        // If we want empty elements for check box
+        if (options.empty && element.type === 'checkbox' && !element.checked) {
+            val = "";
```
Comment (defunctzombie) [inline index.js]: single quotes is the norm in this file

**Comment by defunctzombie** [comment]
Few minor mostly stylistic comments. Really nice work and detailed explanation in the other PR.

**Comment by veacks** [comment]
All done

**Comment by veacks** [comment]
Commits are already squashed.

**Pull request merged**